.gitignore

```diff
@@ -1,6 +1,10 @@
+build*
 *.swp
 *~
 .gdb_history
 /build
 gdb.txt
 /cmake-*
+.idea
```

Comment:
When I opened your project in my IDE, it was always complaining about files being read-only.
What I've done here is letting git ignore the files.
You can still use yours by checking out `.idea` from a previous commit (and not adding it again 😄 )

Reply:
>it was always complaining about files being read-only.

which files are read-only? I don't understand? :)

>You can still use yours by checking out .idea from a previous commit (and not adding it again...)

...on all my various computers? And then put instructions to all IDE users "hey if you're using CLion, please make sure you set these settings else things won't work the way you'd expect, oh and debug support you'll need to pass these environment variables etc..."  At least IntelliJ projects are designed for you to check in their configuration so all the shared settings are...shared. There's a .gitignore file in the .idea directory that ensures per-user configuration files are not added.

Reply:
> which files are read-only? I don't understand? :

When I opened your project and wanted to modify the cmake scripts, Pycharm was complaining about not being able to modify files. The files weren't read-only per se, Pycharm was always wanting me to confirm to modify a non-project file.
)

> ...on all my various computers? And then put instructions to all IDE users "hey if you're using CLion, please make sure you set these settings else things won't work the way you'd expect, oh and debug support you'll need to pass these environment variables etc..." At least IntelliJ projects are designed for you to check in their configuration so all the shared settings are...shared. There's a .gitignore file in the .idea directory that ensures per-user configuration files are not added.

Oh, I didn't know about this feature.
I was assuming the project files got committed by accident.
I guess it will be conflict between different families of IntelliJ products.
I'll revert that patch.

Reply:
After revert, this is the warning that I get shown when I want to modify a source.
![Screenshot from 2020-08-03 23-23-41](https://user-images.githubusercontent.com/4138939/89228735-75904080-d5e0-11ea-9de5-cd9c501825cc.png)

Reply:
I can fix it by explicitly adding seasocks to the *Content root* of Seasocks.
(Setting >> Project: Seasocks >> Project Structure >> Add Content Root)

Reply:
that seems like a pycharmish issue: CLion is also by jetbrains and uses the same project files, but of course pycharm doesn't understand the C++ things in the project. Thus it's confused... I'm not too worried about this use case: I don't think many folks will try and open a C++ project in a python IDE and expect it to work. But obviously you did :grin: so if we can add the content root and it doesn't break CLion, then that's cool. Or we can remove the `.idea` directory; in this particular case there's nothing too difficult to set up. I was more interested in the general case. The JB IDEs specifically go out of their way to try and make checking in and sharing IDE settings low friction (it's their default behaviour).